src/main/osd/osd_elements.c

```diff
@@ -993,28 +997,28 @@ static void osdElementFlymode(osdElementParms_t *element)
     //  5. ACRO
 
     if (FLIGHT_MODE(FAILSAFE_MODE)) {
-        strcpy(element->buff, "!FS!");
+        strcpy(element->buff, STR_ODSE_FLYMODE_FAILSAFE);
```

Comment:
You can still use the constants. Simply wrap the define with `STR(X)` where X is the string (but without the quotes).

Reply:
I did't get this, you mean:
 in localisation.c
#define aSTR(x) const char STR_##x[] = TR_##x     and     aSTR(ODSE_FLYMODE_FAILSAFE);
in localisation.h
#define eSTR(x) extern const char STR_##x[] = TR_##x     and     eSTR(ODSE_FLYMODE_FAILSAFE);
#define STR(x) STR_##x
and in osd_element.c 
strcpy(element->buff, STR(ODSE_FLYMODE_FAILSAFE));

Reply:
So rather than all the constant arrays.

you can:

```
#define ODSE_FLYMODE_FAILSAFE       !FS!

strcpy(element->buff, STR(ODSE_FLYMODE_FAILSAFE));
```

i.e. you can use the define directly - and it merely replaces the inline string that existed before.

Reply:
I guess I am trying to understand the purpose of translations.c?

Reply:
the idea came from firmware to radios (OpenTx and EdgeTx), where you keep size of compiled image low by the possibility to reuse (translated) strings.
localications.c are the storage for those strings and each one of them initilaized by the TRxx equivalent.

Reply:
Regarding automatic update.
I think, in case of firmware development where size and speed matters, the developer need to be in charge of changes, and in the way of PRs. Fx the function cliPrintInternal() made the pgm to went down when met a char STR_XX[] variable, and this kind of issues are not a job for a translator to discover in firmware.
So to use Crowdin to translation, which are smart, I think a tool who take TR* in location.c as key for lookup in crowdin XML file and get locale and translated string to write/replace in xx/locale.h will be the best way.
In Crowdin the key then will be TR_something and it's translated string.

Reply:
But we are using definitions, so there is no "reuse" per se. i.e. based on the selected locale at compile time, it will only use those locale translations - all the others are discarded.

I would get it if there was a need to be able to switch translation at runtime.

Reply:
hmm, you got a point, and in EdgeTx you can change in runtime.
Well I agree and change - will take a while .-)

Reply:
Still working.
Reduced to only defines, but now have an bug to find.

Reply:
Now done, hop its well